order/src/main/java/com/yas/order/service/CheckoutService.java

```diff
@@ -33,70 +35,94 @@
 @Transactional
 @RequiredArgsConstructor
 public class CheckoutService {
+
     private final CheckoutRepository checkoutRepository;
-    private final CheckoutItemRepository checkoutItemRepository;
     private final OrderService orderService;
+    private final ProductService productService;
     private final CheckoutMapper checkoutMapper;
 
     public CheckoutVm createCheckout(CheckoutPostVm checkoutPostVm) {
```

Comment:
Hi @minhtridn2001, @NhatTranMinh15,

Do you think this method should be named `initCheckout(...)`? It seems to handle the case when the user initiates the checkout.

If this method is meant to create a `checkout`, it should handle more cases than just the `PENDING` status.

Reply:
Moreover, should we check if the `checkout` already exists (by verifying the `checkout status` and `checkout items`) instead of creating a new one each time the user clicks **Proceed to checkout**, especially when they might click it multiple times with the same items?

Reply:
> Do you think this method should be named initCheckout(...)?

I think `createCheckout()` makes more sense since the checkout object is persisted into the db